ocaml/message-switch/async/protocol_async.ml

```diff
@@ -82,13 +82,13 @@ module M = struct
 
     let with_lock t f =
       let rec wait state =
-        if Bool.(t.m = state) then
+        if Bool.(t.m = state) then (
+          t.m <- true ;
```

Comment:
As `state` is a variable in this function, while called with `false` below, I think this line should be 

    t.m <- not state ;

Reply:
This is only ever called with `state = false`, I changed it to take a `()`, and that actually simplified this line too:
``` diff
-          Condition.wait t.c >>= fun () -> wait state
+          Condition.wait t.c >>= wait
```